### PR TITLE
SYNCT-75: Work around for comparing FHIR Patient objects.

### PR DESCRIPTION
**Pull request by pkornowski**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir/api/util/FHIRPatientUtil.java [hunk 1]
```diff
@@ -339,4 +339,73 @@ public static Reference buildPatientReference(org.openmrs.Patient patient) {
...
+		Patient p1 = (Patient) patient1;
+		Patient p2 = (Patient) patient2;
+
+		if (p1.getIdentifier().size() == p2.getIdentifier().size()) {
+			for (int i = 0; i < p1.getIdentifier().size(); i++) {
+				if (!p1.getIdentifier().get(i).equalsDeep(p2.getIdentifier().get(i))) {
+					return false;
+				}
+			}
+		}
```
Comment (dserkowski) [inline api/src/main/java/org/openmrs/module/fhir/api/util/FHIRPatientUtil.java]: I think that there should be 'else' statement with 'return false;' line

File: api/src/main/java/org/openmrs/module/fhir/api/util/FHIRPatientUtil.java [hunk 2]
```diff
@@ -339,4 +339,73 @@ public static Reference buildPatientReference(org.openmrs.Patient patient) {
...
+
+		if (p1.getName().size() > 0 && p2.getName().size() > 0) {
```
Comment (dserkowski) [inline api/src/main/java/org/openmrs/module/fhir/api/util/FHIRPatientUtil.java]: Is it the work-around? I think that it should be comment in the code that will explain why you did this in this way

File: api/src/main/java/org/openmrs/module/fhir/api/util/FHIRPatientUtil.java [hunk 3]
```diff
@@ -339,4 +339,73 @@ public static Reference buildPatientReference(org.openmrs.Patient patient) {
...
+			}
+		}
+
+		if (null != p1.getMaritalStatus()) {
+			if (!p1.getMaritalStatus().equalsDeep(p2.getMaritalStatus())) {
+				return false;
+			}
+		}
+
+		if (null != p1.getMultipleBirth()) {
```
Comment (dserkowski) [inline api/src/main/java/org/openmrs/module/fhir/api/util/FHIRPatientUtil.java]: I have noticed that FHIR Patient domain object has more fields (for example birthDate). Shouldn't we compare them?

Comment (pkornowski) [reply api/src/main/java/org/openmrs/module/fhir/api/util/FHIRPatientUtil.java]: There is still the problem like on the REST API with birthDate. I think this is related with: https://issues.openmrs.org/browse/SYNCT-91

**Pull request merged**